framework/tlvf/AutoGenerated/include/tlvf/ieee_1905_1/tlvDeviceInformation.h

```diff
@@ -63,9 +53,37 @@ class tlvDeviceInformation : public BaseClass
         eTlvType* m_type = nullptr;
         uint16_t* m_length = nullptr;
         sMacAddr* m_mac = nullptr;
-        uint8_t* m_info_length = nullptr;
-        sInfo* m_info = nullptr;
-        size_t m_info_idx__ = 0;
+        uint8_t* m_local_interface_list_length = nullptr;
+        cLocalInterfaceInfo* m_local_interface_list = nullptr;
+        size_t m_local_interface_list_idx__ = 0;
+        std::vector<std::shared_ptr<cLocalInterfaceInfo>> m_local_interface_list_vector;
+        bool m_lock_allocation__ = false;
+        int m_lock_order_counter__ = 0;
+};
+
+class cLocalInterfaceInfo : public BaseClass
+{
+    public:
+        cLocalInterfaceInfo(uint8_t* buff, size_t buff_len, bool parse = false);
+        cLocalInterfaceInfo(std::shared_ptr<BaseClass> base, bool parse = false);
+        ~cLocalInterfaceInfo();
+
+        sMacAddr& mac();
+        eMediaType& media_type();
+        uint8_t& media_info_length();
+        uint8_t* media_info(size_t idx = 0);
+        bool alloc_media_info(size_t count = 1);
+        void class_swap() override;
+        bool finalize() override;
+        static size_t get_initial_size();
+
+    private:
+        bool init();
+        sMacAddr* m_mac = nullptr;
+        eMediaType* m_media_type = nullptr;
+        uint8_t* m_media_info_length = nullptr;
+        uint8_t* m_media_info = nullptr;
+        size_t m_media_info_idx__ = 0;
```

Comment:
Nitpick: why the `__` ending? The `m_` prefix already indicates it's a member variable ...